app/workers/extract_worker.rb

```diff
@@ -28,5 +24,20 @@ def perform(classification_or_legacy_workflow_id, legacy_classification_data=nil
         workflow.webhooks.process(:new_extraction, extract.data) if workflow.subscribers?
       end
     end
+  rescue ActiveRecord::RecordNotFound => e
+    if Extract.where(classification_id: classification_id).any?
```

Comment:
I think .exists? is quicker here as it does `limit 1` vs count > 0

Reply:
Both seem to make the same query for me:

```
[5] pry(main)> Extract.exists?
  Extract Exists (0.5ms)  SELECT  1 AS one FROM "extracts" LIMIT $1  [["LIMIT", 1]]
=> true
[6] pry(main)> Extract.any?
  Extract Exists (0.4ms)  SELECT  1 AS one FROM "extracts" LIMIT $1  [["LIMIT", 1]]
=> true
```

Reply:
Intriguing, i get different behaviour when i supply where conditions
```
[3] pry(main)> User.where(id: 4).exists?
    User Exists (0.5ms)  SELECT  1 AS one FROM "users" WHERE "users"."id" = $1 LIMIT 1  [["id", 4]]
=> true
[4] pry(main)> User.where(id: 4).any?
    (0.4ms)  SELECT COUNT(*) FROM "users" WHERE "users"."id" = $1  [["id", 4]]
=> true
```

Reply:
noting that it doesn't matter, as long as the query is fast :)

Reply:
TIL: This seems to be unified in Rails 5.